tests/system/source/main.d

```diff
@@ -88,6 +114,143 @@ void main ()
         checkBlockHeight(1);
     }
 
+    /// Make Transactions for block height 2
+    {
+        API[NodeCnt] clients;
+        foreach (idx, const ref addr; Addrs)
+            clients[idx] = new RestInterfaceClient!API(addr.withSchema());
+
+        // clients[0] Create frozen transactions and enrolment
+        const Block[] blocks_1 = clients[0].getBlocksFrom(1,1);
+        assert(blocks_1[0].header.height == 1);
+
+        foreach (idx; 0 .. Block.TxsInBlock)
+        {
+            Transaction tx;
+
+            if (keypairs.length > idx)
+            {
+                Transaction freeze_tx = {
+                    type: TxType.Freeze,
+                    inputs: [Input(hashFull(blocks_1[0].txs[idx]), 0)],
+                    outputs: [Output(Amount.MinFreezeAmount,
+                        keypairs[idx].address)]
+                };
+                freeze_tx.inputs[0].signature = kp.secret.sign(hashFull(freeze_tx)[]);
+                clients[0].putTransaction(freeze_tx);
+                freeze_txs ~= freeze_tx;
+            }
+            else
+            {
+                Transaction payment_tx = {TxType.Payment, [], []};
+                payment_tx.inputs ~= Input(hashFull(blocks_1[0].txs[idx]), 0);
+                foreach (idx2; 0 .. Block.TxsInBlock)
```

Comment:
It took me a long time to understand what was happening here. I suggest adding a comment:

/// 
```suggestion
                // create tx's to be spent by all tx's in block 3
                foreach (idx2; 0 .. Block.TxsInBlock)
```

Also, the tests only use the transaction in the first `payment_txs` array. I understand why it was written this way, it is just more complicated to reason about. It would make more sense to have a single `payment_tx` in place of `payment_txs`, and just do `if (payment_tx == Transaction.init) { /// initialize it }`

Reply:
applied the suggestion.

Reply:
> Also, the tests only use the transaction in the first payment_txs array. I understand why it was written this way, it is just more complicated to reason about. It would make more sense to have a single payment_tx in place of payment_txs, and just do if (payment_tx == Transaction.init) { /// initialize it }

I didn't understand the meaning of using single `payment_tx`. 
Could you tell me more?

Reply:
You can keep it the way you wrote it. 

But what I originally meant was:

```diff
- Transaction[] payment_txs;
+ Transaction payment;
```

And:

```diff
- payment_txs ~= payment_tx;
+ if (payment == Transaction.init)
    payment = payment_tx;
```

Because only one payment UTXOs are later used. It's alright to keep it your way if you want to.

Reply:
Thank you. applied the suggestion.